templates/preview.html

```diff
@@ -0,0 +1,133 @@
+{% extends 'layouts/_base.html' %}
+
+{% from "components/panel/_macro.njk" import onsPanel %}
+{% from "components/button/_macro.njk" import onsButton %}
+{% from "components/accordion/_macro.njk" import onsAccordion %}
+
+{% set save_on_signout = true %}
+
+{% set breadcrumbs = {
+  "ariaLabel": 'Back',
+  "itemsList": [
+    {
+      "url": url_for("questionnaire.get_questionnaire"),
+      "id": "top-previous",
+      "text": _("Back"),
+      "attributes": {
+        "data-ga": 'click',
+        "data-ga-category": 'Navigation',
+        "data-ga-action": 'Previous link click'
+      }
+    }
+  ]
+} %}
+
+{% block main %}
+
+  <h1 class="ons-u-mt-m">{{ _("Preview of the questions in this survey") }}</h1>
+
+  {% call onsPanel({
+    "classes": 'ons-u-mb-m ons-u-ph'
+  }) %}
+    <p>{{ _("To answer these questions you need to <a href='{url}'>start survey</a>").format(url=url_for('questionnaire.get_questionnaire')) }}</p>
+  {% endcall %}
+  <p>{{ _("You may not have to answer all of these questions. The questions you see will depend on the answers you provide.") }}</p>
+
+  {{
+    onsButton({
+      "type": 'button',
+      "text": _('Print answers'),
+      "variants": ['small', 'secondary', 'print'],
+      "attributes": {
+        "data-qa": "btn-print",
+        "data-ga-category": "Print button",
+        "data-ga-action": "Open print Dialogue",
+        "data-ga-label": "Print button click"
+      }
+    })
+  }}
+  {{
+    onsButton({
+      "text": _('Save answers as PDF'),
+      "variants": ['small', 'secondary', 'timer', 'download'],
+      "url": content.pdf_url,
+      "removeDownloadAttribute": true,
+      "attributes": {
+        "data-qa": "btn-pdf",
+        "data-ga-category": "PDF button",
+        "data-ga-action": "Download PDF",
+        "data-ga-label": "PDF button click"
+      }
+    })
+  }}
+  <div class="ons-u-mt-l ons-u-mb-l">
+
+    {%- if content.preview.sections | length > 1 -%}
+      {%- set itemList = [] -%}
+      {%- for section in content.preview.sections -%}
+        {%- if section["blocks"] | length -%}
+          {%- set item = {
+            "title": section.title,
+            "id": section.id,
+            "content": "",
+            "button": {
+              "open": _('Show'),
+              "close": _('Hide'),
+              "attributes": {
+                "data-ga": "click",
+                "data-ga-category": "Preview Survey",
+                "data-ga-action": "Open panel",
+                "data-ga-label": section.title
+              }
+            }
+          } -%}
+
+          {%- set section %}
```

Comment:
This is prone to error and shouldn't be same name as the thing we are looping over which is also section.